### PR TITLE
Plfm 4247 b

### PR DESCRIPTION
**Pull request by john-hill**

### CODE REVIEW

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/table/TableEntityManagerImpl.java [hunk 1]
```diff
@@ -992,20 +991,26 @@ void linkVersionToTransaction(String tableIdString, long version, long transacti
...
-	public SnapshotResponse createTableSnapshot(UserInfo userInfo, String tableIdString, SnapshotRequest request) {
+	public SnapshotResponse createTableSnapshot(UserInfo userInfo, String tableId, SnapshotRequest request) {
 		ValidateArgument.required(userInfo, "userInfo");
-		ValidateArgument.required(tableIdString, "TableId");
+		ValidateArgument.required(tableId, "TableId");
 		ValidateArgument.required(request, "request");
-		Long tableId = KeyFactory.stringToKey(tableIdString);
-		IdAndVersion idAndVersion = IdAndVersion.newBuilder().setId(tableId).build();
+		
+		IdAndVersion idAndVersion = KeyFactory.idAndVersion(tableId);
```
Comment (john-hill) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/table/TableEntityManagerImpl.java]: Duplicate of IdAndVersion.parse().

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/entity/ReplicationManagerImpl.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+	 * 
+	 */
+	@Override
+	public void replicate(String entityId) {
+		ValidateArgument.required(entityId, "EntityId");
+		final List<EntityDTO> entityDTOs = nodeDao.getEntityDTOs(Collections.singletonList(entityId),
+				MAX_ANNOTATION_CHARS);
+		// Connect only to the index for this table
+		final TableIndexDAO indexDao = connectionFactory.getConnection(KeyFactory.idAndVersion(entityId));
+		indexDao.executeInWriteTransaction((TransactionStatus status) -> {
```
Comment (john-hill) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/entity/ReplicationManagerImpl.java:126]: extract to its own method.

File: services/repository-managers/src/test/java/org/sagebionetworks/repo/manager/entity/ReplicationManagerTest.java [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+import org.sagebionetworks.repo.model.message.ChangeType;
+import org.sagebionetworks.repo.model.table.EntityDTO;
+import org.sagebionetworks.table.cluster.ConnectionFactory;
+import org.sagebionetworks.table.cluster.TableIndexDAO;
+import org.sagebionetworks.workers.util.aws.message.RecoverableMessageException;
+import org.springframework.transaction.TransactionStatus;
+import org.springframework.transaction.support.TransactionCallback;
+
+import com.google.common.collect.Lists;
+
```
Comment (john-hill) [inline services/repository-managers/src/test/java/org/sagebionetworks/repo/manager/entity/ReplicationManagerTest.java]: convert to junit 5

File: services/repository-managers/src/main/resources/private/managers-spb.xml [hunk 4]
```diff
@@ -522,6 +522,10 @@
 	<bean id="replicationMessageManagerAsynch"
```
Comment (john-hill) [inline services/repository-managers/src/main/resources/private/managers-spb.xml]: use service annotation

**Review by marcomarasca — APPROVED** [review]

**Pull request merged**